templates/OperationResponseJsonBodySnippet.stencil

```diff
@@ -1,5 +1,23 @@
-let decoder = JSONDecoder()
-let decoded = try decoder.decode({{ response.objectType }}.self, from: data)
-dispatchQueue.async {
-    completionHandler(.success(decoded), httpResponse)
+{% include "OperationResponseNullableBodySnippet.stencil" %}
+do {
+    let decoder = JSONDecoder()
+    {% if response.strategy == "dateTimeBody" %}
+    let dateFormatter = Date.AzureISO8601DateFormatter()
+    decoder.dateDecodingStrategy = .formatted(dateFormatter)
+    {% elif response.strategy == "dateTimeRfc1123Body" %}
+    let dateFormatter = Date.AzureRfc1123DateFormatter()
+    decoder.dateDecodingStrategy = .formatted(dateFormatter)
+    {% elif response.strategy == "dateBody" %}
+    let dateFormatter = DateFormatter()
+    dateFormatter.dateFormat = "yyyy-MM-dd"
+    decoder.dateDecodingStrategy = .formatted(dateFormatter)
+    {% endif %}
+    let decoded = try decoder.decode({{ response.objectType }}.self, from: data)
```

Comment:
Would recommend indentation here to make the template more readable. Swiftlint or swiftformat will ultimately fix the generated indentation anyhow.

Reply:
Done